### ISSUE TITLE
[tt-train] layernorm_bw computes wrong dx: mean(dy*gamma) subtracted twice; non-L1 path loses reduction blocks to nested tile_regs_acquire

### ISSUE THREAD

**Issue by ayerofieiev-tt** [tenstorrent/tt-metal#53204]
Found in a systematic audit of tt-train's fused metal kernels.

## Bug 1: `compute_dx` subtracts `mean(dy·γ)` twice (all shapes, all paths)

`tt-train/sources/ttml/metal/ops/layernorm_bw/device/kernels/compute/layernorm_bw_kernel.cpp:416-448`

On WH/BH, `ELWMUL` (`mul_tiles`) always accumulates into DEST (`tt_metal/hw/inc/api/compute/eltwise_binary.h:97-98`; standard LLK path issues no ZEROACC for `ReuseDestType::NONE`). The sequence in `compute_dx`:

1. `copy_tile(s1 → temp_register)`
2. `sub` (dx −= temp)
3. `mul_tiles(cb_x_hat, cb_s2, …, temp_register)` — **accumulates onto temp, which still holds s1**

So temp becomes `s1 + x̂·s2` and the kernel produces:

```
dx = rstd·(dy·γ − 2·mean(dy·γ) − x̂·mean(dy·γ·x̂))     // wrong
dx = rstd·(dy·γ −   mean(dy·γ) − x̂·mean(dy·γ·x̂))     // reference
```

The kernel already guards this exact hazard with `zero_dst_reg()` at lines 179, 353, and 420 — this one call site is missing the zero (or needs reordering).

**Impact:** every training step where `dy·γ` has nonzero feature-mean gets a systematically wrong input gradient. Silent — loss still decreases.

## Bug 2: nested `tile_regs_acquire` destroys the running sum in `compute_dy_gamma_xnorm_sum` (non-L1 path)

Same file, line 329 acquires DEST, then inside the loop (336-337) `compute_x_hat_preprocessing` does its own acquire/commit/`pack_and_push_block`. Every inner release makes the packer zero the DEST half holding the outer accumulation (`llk_pack_common.h:57-70`), so `cb_scaled_dy_gamma_xnorm_sum` holds only the **last block's** contribution to `mean(dy·γ·x̂)`. Triggers whenever `EVERYTHING_FITS_IN_L1` is off (hidden dim ≳ 5000). Nested acquire is also an API protocol violation — it avoids hanging only because acquire depth never exceeds the semaphore capacity of 2. The L1 path precomputes x_hat outside any acquire (504-509) and is fine.

## Why CI is green

`tt-train/tests/ops/layernorm_bw_fused_op_test.cpp:178-180` uses `allclose(rtol=1e-3, atol=5e-1)` on O(1)-magnitude gradients — an absolute error of 0.5 always passes. With uniform random test data, `mean(dy·γ)` ≈ 0.01–0.13, far below the 0.5 atol. The suite proves "not total garbage," nothing more.

## Fix

1. Zero `temp_register` before the `mul_tiles` in `compute_dx` (or reorder to avoid the reuse).
2. Restructure the non-L1 xnorm sum to hoist x_hat computation out of the accumulation acquire window (mirror the L1 path).
3. Tighten bw test atol to ~1e-2 — the suite should fail before the fixes and pass after.

Related cheap fix while in there: fw `cb_sum`/`cb_variance_sum` are declared bf16 but sized fp32 (`layernorm_fw_program_factory.cpp:288-298`) — partial sums round to bf16 before the matmul reduction; bw already uses Float32 correctly.
